onnxruntime/core/graph/contrib_ops/bert_defs.cc

```diff
@@ -377,6 +377,27 @@ ONNX_MS_OPERATOR_SET_SCHEMA(FastGelu, 1,
                                   return true;
                                 }));
 
+ONNX_MS_OPERATOR_SET_SCHEMA(RelPosAttnBiasGen, 1,
+                            OpSchema()
+                                .SetDoc("Compute binned relative position bias")
+                                .Attr("max_distance", "Max distance", AttributeProto::INT)
+                                .Attr("is_bidirectional", "Default value is 0.", AttributeProto::INT, static_cast<int64_t>(0))
+                                .Input(0, "bias_table", "2D input tensor with shape (num_buckets, num_heads), COL-major(See UT for example)", "T")
+                                .Input(1, "sequence_length", "Real sequence length from last dimension of hidden states", "U")
+                                .Output(0, "output", "4D output tensor with shape (1, num_heads, sequence_length, sequence_length)", "T")
```

Comment:
If first dimension is 1, it cannot be feed to extra_add_qk of attention since that requires batch_size as dim 0. What's your design to use this in Attention op?

Reply:
Could you also take a look at the shape of rel_pos_bias for fused fmha kernel? If that also requires batch as first dimension, I think we shall add broadcast here instead of doing it by extra nodes or broadcast in each Attention node. Otherwise, we can keep current design for easy integration of fused kernel.

Reply:
sounds good

Reply:
I took a quick look at fused kernel. It seems that it is for windows attention, which has shape [num_heads, window_size, window_size]
https://github.com/NVIDIA/FasterTransformer/blob/b672f49e256ba7a2d4fc9691d270b60b7fc1a2ff/src/fastertransformer/layers/attention_layers/WindowAttention.cc#L125.
So, let's use (1, num_heads, sequence_length, sequence_length) as shape, and update attention to support such shape for extra_add_qk.

Reply:
> Could you also take a look at the shape of rel_pos_bias for fused fmha kernel? If that also requires batch as first dimension, I think we shall add broadcast here instead of doing it by extra nodes or broadcast in each Attention node. Otherwise, we can keep current design for easy integration of fused kernel.

seems like fmha is not supported: https://github.com/NVIDIA/FasterTransformer/blob/main/src/fastertransformer/tf_op/t5/T5EncoderOp.cc#L234
I think those bias related fmha is for WindowAttention